### PR TITLE
Update providers of virtual packages related to Intel OneAPI

### PR DESCRIPTION
**Pull request by hpcnpatel**
This PR brings, 

- Update default providers of virtual packages from `intel-<component>` to `intel-oneapi-<component>`.

- Replace specific name of `intel-oneapi-tbb` to a virtual package `tbb` in the depend_on().

### CODE REVIEW

File: etc/spack/defaults/packages.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ packages:
       scalapack: [netlib-scalapack, amdscalapack]
       sycl: [hipsycl]
       szip: [libaec, libszip]
-      tbb: [intel-tbb]
+      tbb: [intel-oneapi-tbb]
```
Comment (rscohn2) [inline etc/spack/defaults/packages.yaml]: intel-tbb is built from open source. intel-oneapi-tbb is a binary-only distribution. It is based on open source, but not guaranteed to be the same. I think some people would object to making intel-oneapi-tbb the default provider so I prefer not to change this.

intel-daal, intel-ipp, intel-mkl are binary-only distributions. They have not been updated for 3 years so it makes sense to move forward.

File: etc/spack/defaults/packages.yaml
Comment (rscohn2) [inline etc/spack/defaults/packages.yaml:1]: This PR changes the default provider for mkl, daal, & ipp. The current default provider is from parallel studio, which has not been updated in 3 years. The new default provider would come from oneapi. Below, I will tag the impacted package maintainers to get feedback:
ipp:
* opencv: @bvanessen @adamjstewart @glennpj
* py-accimage: no maintainers: not conditional on a variant 

daal: no virtual uses

mkl:
* abacus: @Bitllion: not conditional on a variant 
* amgx: @js947 
* eckit: @skosukhin 
* elk: no maintainer. Expects `threads=openmp` variant
* exciting: no maintainer
* fplo: @glennpj : not conditional on a variant 
* hpcc: no maintainer
* ldak: no maintainer
* ligra: no maintainer
* namd: no maintainer
* petsc: @balay @BarrySmith @jedbrown 
* povray: no maintainer
* psrchive: no maintainer
* py-tensorflow: @adamjstewart @aweits 
* py-torch: @adamjstewart 
* relion: @dacolombo 
* shengbte: no maintainer: not conditional on a variant 
* votca-tools: @junghans 
* votca: @junghans

File: etc/spack/defaults/packages.yaml (hunk 1 above)
Comment (hpcnpatel) [reply etc/spack/defaults/packages.yaml]: i agree with your comments about not changing the oneapi to be the default provider of tbb. I will revert the change.

**Comment by rscohn2** [comment]
From trying out installs and inspection, I see that this needs to be updated to oneapi package layout: https://github.com/spack/spack/blob/13b0b1d574802c6534814d46ccc54038b6bc000f/var/spack/repos/builtin/packages/elk/package.py#L174

I will add it to your commit. Other packages fail to build, but it looks unrelated to the change.

**Review by rscohn2 — APPROVED** [review]
Thanks!

**Pull request merged**